cmd/statusd-cli/repl.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+)
+
+// REPL implements the read-eval-print loop for the commands
+// to be sent to statusd.
+type REPL struct {
+	conn   net.Conn
+	reader *bufio.Reader
```

Comment:
Could it be an interface like `io.Reader` or `io.ReadCloser`?

Reply:
`net.Conn` already implements `io.ReadWriteCloser`. The `bufio` types simply help reading and writing `\n` terminated strings.

Reply:
Ok, I see for example `r.reader.ReadString('\n')` and `ReadString` is not a part of `io.Reader` interface, that's why you need a concrete type.